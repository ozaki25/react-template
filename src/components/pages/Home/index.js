import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import logo from './logo.svg';
import './index.css';

class Home extends React.Component {
  constructor(props) {
    super(props);
    this.input = React.createRef();
  }

  sayHello = () => {
    const input = this.input.current.value;
    this.input.current.value = '';
    this.props.greet(input);
  };

  removeHello = (e) => {
    this.props.bye(Number(e.target.dataset.index));
  };

  render() {
    const { greetings, getWeather, weather } = this.props;
    const { loading, body, error } = weather;
    const { description } = body;
    return (
      <div className="Home">
        <header className="Home-header">
          <img src={logo} className="Home-logo" alt="logo" />
          <h1 className="Home-title">Welcome to React</h1>
        </header>
        <p className="Home-intro">
          To get started, edit <code>src/Home.js</code> and save to reload.
        </p>
        <Link to="/hello">Hello!</Link>
        {/* redux検証用 */}
        <div>
          <input ref={this.input} />
          <button type="button" onClick={this.sayHello}>
            Hello
          </button>
          {greetings.map((greet, i) => (
            // eslint-disable-next-line  react/no-array-index-key
            <p key={i}>
              Hello {greet}
              <a href="#" data-index={i} onClick={this.removeHello}>
                &times;
              </a>
            </p>
          ))}
        </div>
        <div>
          <button type="button" onClick={getWeather}>
            天気
          </button>
          <h1>{loading ? 'Loading...' : ''}</h1>
          <h2>{body.title}</h2>
          <pre>{description ? description.text : ''}</pre>
          <p>{error ? error.message : ''}</p>
        </div>
      </div>
    );
  }
}

Home.propTypes = {
  greetings: PropTypes.arrayOf(PropTypes.string).isRequired,
  greet: PropTypes.func.isRequired,
  bye: PropTypes.func.isRequired,
  weather: PropTypes.shape({
    loading: PropTypes.bool,
    body: PropTypes.object,
    error: PropTypes.object,
  }).isRequired,
  getWeather: PropTypes.func.isRequired,
};

export default Home;
