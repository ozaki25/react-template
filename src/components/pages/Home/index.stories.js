import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import Home from './';

const stories = storiesOf('pages', module);
stories.addDecorator(story => <BrowserRouter>{story()}</BrowserRouter>);
stories.add('Home', withInfo('Home')(() => <Home />));
