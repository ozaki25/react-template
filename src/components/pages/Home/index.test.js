import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { snapshot } from 'test/helpers';
import Home from '.';

const props = {
  greetings: [],
  greet: jest.fn(),
  bye: jest.fn(),
  weather: {
    loading: false,
    body: {
      title: '東京都の天気',
      description: {
        text: '晴れ',
      },
    },
    error: {},
  },
  getWeather: jest.fn(),
};

snapshot(
  'Hello',
  <BrowserRouter>
    <Home {...props} />
  </BrowserRouter>,
);
