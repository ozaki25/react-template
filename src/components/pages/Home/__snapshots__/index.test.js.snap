// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: Hello 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Home
      bye={[Function]}
      getWeather={[Function]}
      greet={[Function]}
      greetings={Array []}
      weather={
        Object {
          "body": Object {
            "description": Object {
              "text": "晴れ",
            },
            "title": "東京都の天気",
          },
          "error": Object {},
          "loading": false,
        }
      }
    >
      <div
        className="Home"
      >
        <header
          className="Home-header"
        >
          <img
            alt="logo"
            className="Home-logo"
            src="logo.svg"
          />
          <h1
            className="Home-title"
          >
            Welcome to React
          </h1>
        </header>
        <p
          className="Home-intro"
        >
          To get started, edit 
          <code>
            src/Home.js
          </code>
           and save to reload.
        </p>
        <Link
          replace={false}
          to="/hello"
        >
          <a
            href="/hello"
            onClick={[Function]}
          >
            Hello!
          </a>
        </Link>
        <div>
          <input />
          <button
            onClick={[Function]}
            type="button"
          >
            Hello
          </button>
        </div>
        <div>
          <button
            onClick={[Function]}
            type="button"
          >
            天気
          </button>
          <h1 />
          <h2>
            東京都の天気
          </h2>
          <pre>
            晴れ
          </pre>
          <p />
        </div>
      </div>
    </Home>
  </Router>
</BrowserRouter>
`;
